start_bundle_coupled.R

```diff
@@ -260,61 +267,65 @@ for(scen in common){
       already_rem <- mixedsort(Sys.glob(suche))[1]
     }
   }
-  if (is.na(already_rem)) {
-    message("Nothing found for ", suche, ", starting with ", runname, "-rem-1.")
-  } else {
+  
+  if (.isFileAndAvailable(settings_remind[scen, "path_gdx"], "/fulldata.gdx")) {
+    # if there is a REMIND gdx given in scenario_cofig or scenario_config_coupled, use it instead of the one found automatically
+    message("Using REMIND gdx specified in ", basename(path_settings_coupled)," or ", basename(path_settings_remind),": ", settings_remind[scen, "path_gdx"])
+  } else if (! is.na(already_rem)) {
     # if there is an existing REMIND run, use its gdx for the run to be started
     settings_remind[scen, "path_gdx"] <- normalizePath(already_rem)
     message("Found REMIND gdx here: ", normalizePath(already_rem))
+  } else {
+    message("Nothing found for ", suche, ", starting with ", runname, "-rem-1.")
   }
+  
   # is there already a MAgPIE run with this name?
   iter_mag <- 0
-  needle <- scen
-  suche <- paste0(path_magpie, "output/", prefix_runname, needle,"-mag-*/report.mif")
+  suche <- paste0(path_magpie, "output/", prefix_runname, scen,"-mag-*/report.mif")
   already_mag <- mixedsort(Sys.glob(suche))[1]
+  
   if (! is.na(already_mag)) {
     iter_mag <- as.integer(sub(".*mag-(\\d.*)/.*","\\1",already_mag))
-  } else if (! is.na(scenarios_coupled[scen, "oldrun"])) {
-    message("Nothing found for ", suche, ", continue with oldrun")
-    if (isTRUE(stringr::str_sub(scenarios_coupled[scen, "oldrun"], -14, -1) == "/report.mif") &&
-        file.exists(scenarios_coupled[scen, "oldrun"])) {
-          already_mag <- c(scenarios_coupled[scen, "oldrun"])
+  } else {
+    message("Nothing found for ", suche, ", continue with path_report")
+    if (.isFileAndAvailable(scenarios_coupled[scen, "path_report"], "/report.mif")) {
+          already_mag <- scenarios_coupled[scen, "path_report"]
     } else {
-      needle <- scenarios_coupled[scen, "oldrun"]
-      suche <- paste0(path_magpie_oldruns, prefix_oldruns, needle, "-mag-*/report.mif")
+      message("Nothing found for path_report continue with oldrun")
+      lookfor <- if (is.na(scenarios_coupled[scen, "oldrun"])) scen else scenarios_coupled[scen, "oldrun"]
+      suche <- paste0(path_magpie_oldruns, prefix_oldruns, lookfor, "-mag-*/report.mif")
       already_mag <- mixedsort(Sys.glob(suche))[1]
     }
   }
+
+  path_report_found <- NULL
   if (is.na(already_mag)) {
     message("Nothing found for ", suche, ", starting REMIND standalone.")
   } else {
     path_report_found <- normalizePath(already_mag)
     message("Found MAgPIE report here: ", path_report_found)
   }
+  
   # decide whether to continue with REMIND or MAgPIE
+  start_magpie <- FALSE
   if (iter_rem == iter_mag + 1 & iter_rem < max_iterations) {
     # if only remind has finished an iteration -> start with magpie in this iteration using a REMIND report
     start_iter_first  <- iter_rem
```

Comment:
I'm not sure whether `already_rem` is correct here in line 332. In the way you reformulated it, this may not even be defined, right? Or it may point to a run that isn't actually used because you specified a `…/fulldata.gdx` in `path_gdx`. I guess it must be `settings_remind[scen, "path_gdx"]`

Reply:
Interesting. If I am not mistaken the question whether it is defined or not has not been touched by the changes. If `already_rem` is not defined (no former REMIND run found), `iter_rem` will be zero and this line will not be executed.

Reply:
That is true. The problem is instead: This code will be executed only if it doesn't start the first REMIND, but say `rem-2`. So `already_rem` is defined (as it was finding old paths), but you now don't like to use this one, but the one specified in the config file… And so it should catch a wrong one, doesn't it?